### PR TITLE
chore: ensure CLI works with other key formats

### PR DESCRIPTION
**Pull request by big-andy-coates**
### Description 

fixes: https://github.com/confluentinc/ksql/issues/6218

Updates a few places in the rest api, cli and the new Java client to specify the key format as well as the value format.

Unfortunately, the value format was called `format`.  This change introduces a `keyFormat` and `valueFormat` to hold the two formats, but both the CLI and new Java client are backwards compatible with the old JSON schema.

Also in this commit, streams also indicate if there key is windowed.  This means the key is serialized using the windowed serializer. It is possible to have a stream over a topic with a windowed key. Now the CLI and Java client expose this info.

Main change was to `show tables` and `show streams` response.  Previously, these returned

```
ksql> show streams;

 Stream Name | Kafka Topic | Format
---------------------------------------
 PAGEVIEWS   | pageviews   | DELIMITED
---------------------------------------

ksql> show tables;

 Table Name | Kafka Topic | Format | Windowed
----------------------------------------------
 USERS      | users       | JSON   | false
----------------------------------------------
```

And now return:

```
ksql> show streams;

 Stream Name | Kafka Topic | Key Format | Value Format | Windowed
-----------------------------------------------------------------
 PAGEVIEWS   | pageviews   | KAFKA      | DELIMITED    | false
-----------------------------------------------------------------

ksql> show tables;

 Table Name | Kafka Topic | Key Format | Value Format | Windowed
----------------------------------------------------------------
 USERS      | users       | KAFKA      | JSON          | false
----------------------------------------------------------------
```

### Testing done 

Usual + manually created streams and tables with non-KAFKA keys and ensured it all worked.

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

**Comment by big-andy-coates** [comment]
@vcrfxia, following on from our previous conversation about using the existing rest model pojos and Jackson, rather than new classes and hand crafted JSON deserialization code... 

This PR is a good example of why we should share the model between the server and the java client. Because we don't I needed to update two parallel sets of code that handle the JSON<>Java conversion for the `SourceInfo`, and both bits of code needed to be enhanced and tested to ensure they were backwards compatible.

Over time, this shadow model and deserialization code will be expensive to maintain and invariably get out of sync.

**Review by JimGalasyn — APPROVED** [review]
Excellent!

File: ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/StreamInfoImpl.java [hunk 1]
```diff
@@ -41,8 +51,18 @@ public String getTopic() {
   }
 
   @Override
-  public String getFormat() {
```
Comment (vcrfxia) [inline ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/StreamInfoImpl.java:44]: This is a public API. Should we leave it here and mark as deprecated, to be removed in a future release?

File: ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/TableInfoImpl.java [hunk 2]
```diff
@@ -48,13 +51,18 @@ public String getTopic() {
   }
 
   @Override
-  public String getFormat() {
```
Comment (vcrfxia) [inline ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/TableInfoImpl.java:51]: As above.

**Review by vcrfxia — APPROVED** [review]
LGTM -- thanks! One question inline.

> @vcrfxia, following on from our previous conversation about using the existing rest model pojos and Jackson, rather than new classes and hand crafted JSON deserialization code...
>
> This PR is a good example of why we should share the model between the server and the java client.

Agreed. There's already an issue for this (https://github.com/confluentinc/ksql/issues/6042) and it's marked as a blocker before any additional features are added (or other major changes are made) to the client, such as https://github.com/confluentinc/ksql/issues/6269.

File: ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/StreamInfoImpl.java (hunk 1 above)
Comment (big-andy-coates) [reply ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/StreamInfoImpl.java:44]: My personal view is no, I think the Java client is new enough we can get away with just changing this. People will get a compilation error and will just switch to the new method. But... we don't actually have a defined policy for this yet.

Comment (big-andy-coates) [reply ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/StreamInfoImpl.java:44]: Meh... done.

**Pull request merged**